src/openfl/display/DisplayObject.hx

```diff
@@ -571,7 +571,13 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable {
 	private inline function __setRenderDirty():Void {
 		if (!__renderDirty) {
 			__renderDirty = true;
-			__setParentRenderDirty();
+			if (__maskTarget != null) {
+				// we want to re-render the mask target as well as update it
+				// so it recalculates its `__renderable` considering mask's visibility
+				__maskTarget.__setRenderDirty();
+			} else {
+				__setParentRenderDirty();
+			}
```

Comment:
I like this! Still, with all the flags and updates in OpenFL, this might have some sideeffects, but likely we would find catch them with one of our codebases I believe ;)